File: src/Blazor.Code.Shared/Interaction/CompileInteraction.cs
using Microsoft.Extensions.DependencyInjection;
using Microsoft.JSInterop;

namespace Blazor.Code.Shared;

public static class CompileInteraction
{
    /// <summary>
    /// 编译Razor
    /// </summary>
    /// <param name="code"></param>
    /// <returns></returns>
    [JSInvokable]
    public static async Task CompileRazor(string code)
    {
        var keyEventBus = BlazorCodeSharedExtension.ServiceProvider?.GetRequiredService<KeyEventBus<string>>();

        await keyEventBus?.PushAsync(BlazorCodeConstant.RenderComponent, code);
    }
}

File: web/src/module/interaction/compile.interaction.ts
/**
 * 定义js互操对象（因为在ts不能有未定义但是当前方法是存在windo的）
 */
declare const DotNet;

const assemblyName = 'Blazor.Code.Shared';

class CompileInteraction {

    /**
     * 动态编辑razor
     * @param code 
     */
    async CompileRazor(code: string) {
        await DotNet.invokeMethodAsync(assemblyName, 'CompileRazor', code);
    }
}

// 这样就相当于静态方法全局只有一个
export default new CompileInteraction()